### PR TITLE
#2753 Snap arrows to horizontal and vertical orientation

### LINKED ISSUE #2753: Snap arrows to horizontal and vertical orientation
When user positions an arrow on canvas by dragging one of its ends, it is hard to make it oriented strongly horizontally or vertically.

In order to facilitate horizontal and vertical orientation the arrow should snap to horizontal (0 degrees) position once the angle between the arrow and the horizontal line is 15 degrees or less.  Once the arrow is snapped and while user holds mouse button pressed, the arrow should be highlighted with blue color to indicate horizontal position.
No snapping should be applied if user holds Ctrl key pressed while dragging the arrow. The arrow should be highlighted though in this case if it is positioned exactly to 0 degrees.

Similar behavior should be introduced for vertical (90 degrees) snapping. 

#185260742

### PR DESCRIPTION
**Pull request by yuleicul**
Closes #2753 

### Basic acceptance criteria
- [x] When creating a new _Reaction Arrow_, it snaps and changes color once the angle is 15 degrees or less
    
  arrow-snap-01

- [x] When dragging any end of _Reaction Arrow_ by **Arrow Tool**, it snaps and changes color once the angle is 15 degrees or less

  arrow-snap-02

- [x] When dragging any end of _Reaction Arrow_ by **Select Tool**, it snaps and changes color once the angle is 15 degrees or less

  arrow-snap-03

- [x] No snapping should be applied if `Ctrl` key is pressed, but the arrow should change color

  arrow-snap-04

### CODE REVIEW

File: packages/ketcher-core/src/application/editor/operations/rxn/RxnArrowResize.ts [hunk 1]
```diff
@@ -14,69 +14,106 @@
...
       let middlePoint
-      let reItem
 
       if (RxnArrow.isElliptical(item)) {
-        reItem = restruct.rxnArrows.get(id)
         ;[, , middlePoint] = reItem.getReferencePoints()
       }
 
       if (
+        /**
```
Comment (Nitvex) [inline packages/ketcher-core/src/application/editor/operations/rxn/RxnArrowResize.ts:70]: Proposal:
to draw an image, describe it with words and put it in some kind of README.md file in this folder.
I guess, we can document such a behaviour and complex logic in our application.

File: packages/ketcher-react/src/script/editor/tool/reactionarrow.ts [hunk 2]
```diff
@@ -36,9 +37,16 @@ class ReactionArrowTool implements Tool {
     this.editor.selection(null)
   }
 
+  get render() {
```
Comment (kaluginserg) [inline packages/ketcher-react/src/script/editor/tool/reactionarrow.ts:40]: suggestion: `private` modifier

File: packages/ketcher-react/src/script/editor/tool/reactionarrow.ts [hunk 3]
```diff
@@ -36,9 +37,16 @@ class ReactionArrowTool implements Tool {
...
+    return this.editor.render
+  }
+
+  get reStruct() {
```
Comment (kaluginserg) [inline packages/ketcher-react/src/script/editor/tool/reactionarrow.ts:44]: suggestion: `private` modifier

File: packages/ketcher-core/src/application/render/draw.ts [hunk 4]
```diff
@@ -144,636 +215,745 @@ function arrow(
...
+      `L${tfx(start.x + arrowLen + unbalanceVal)},${tfx(
+        start.y + arrowOffset
       )}` +
-      `L${tfx(point1.x + arrowLen + unbalanceVal)},${point1.y + lineOffset}Z`
+      `L${tfx(start.x + arrowLen + unbalanceVal)},${start.y + lineOffset}Z`
   )
 
   const transformedPath = svgPath(path.join(''))
-    .rotate(arrowAngle, point1.x, point1.y)
+    .rotate(arrowAngle, start.x, start.y)
```
Comment (kaluginserg) [inline packages/ketcher-core/src/application/render/draw.ts:949]: suggestion for the next PR: could you create Refactoring PR separately - it takes time to review each line and distinguesh - what was the function change and what was the rename?

File: packages/ketcher-core/src/application/render/draw.ts [hunk 5]
```diff
@@ -81,43 +81,112 @@ function arrow(
   item: ArrowItem,
   length: number,
   angle: number,
-  options: RenderOptions
+  options: RenderOptions,
+  isResizing: boolean
 ) {
+  const shouldApplySnappingStyle =
+    isResizing &&
+    ['0', '-0', '90', '-90', '180', '-180'].includes(angle.toFixed())
```
Comment (kaluginserg) [inline packages/ketcher-core/src/application/render/draw.ts:89]: '-0' - not sure that js knows about such cases. Suggest removing this item from the array.

File: packages/ketcher-core/src/application/render/draw.ts (hunk 4 above)
Comment (yuleicul) [reply packages/ketcher-core/src/application/render/draw.ts:949]: Sorry for mixing the new feature and refactoring together. Will pay more attention next time, thank you!

Comment (Nitvex) [reply packages/ketcher-core/src/application/render/draw.ts:949]: Or other proposal – you can squash commits and for example, create 2: one for the task itself and another one for refactoring

Comment (yuleicul) [reply packages/ketcher-core/src/application/render/draw.ts:949]: Good proposal, thank you! Done

File: packages/ketcher-core/src/application/editor/operations/rxn/RxnArrowResize.doc.png
Comment (kaluginserg) [inline packages/ketcher-core/src/application/editor/operations/rxn/RxnArrowResize.doc.png:1]: probably there are better place for the documentation /documentation

**Review by kaluginserg — APPROVED** [review]

**Pull request merged**